sympy/functions/elementary/exponential.py

```diff
@@ -576,22 +581,37 @@ def eval(cls, arg, base=None):
                 return S.ComplexInfinity
             elif arg is S.Exp1:
                 return S.One
-            cst_table = {
-                sqrt(3)/2 + I*S.Half: S.Pi/6,
-                sqrt(3)/2 - I*S.Half: -S.Pi/6,
-                sqrt(2)/2 + I*sqrt(2)/2: S.Pi/4,
-                sqrt(2)/2 - I*sqrt(2)/2: -S.Pi/4,
-                S.Half + I*sqrt(3)/2: S.Pi/3,
-                S.Half - I*sqrt(3)/2: -S.Pi/3,
-                -S.Half + I*sqrt(3)/2: 2*S.Pi/3,
-                -S.Half - I*sqrt(3)/2: -2*S.Pi/3,
-                -sqrt(2)/2 + I*sqrt(2)/2: 3*S.Pi/4,
-                -sqrt(2)/2 - I*sqrt(2)/2: -3*S.Pi/4,
-                -sqrt(3)/2 + I*S.Half: 5*S.Pi/6,
-                -sqrt(3)/2 - I*S.Half: -5*S.Pi/6
+
+        if arg.is_number and arg.is_algebraic:
+            # Check for arguments involving rational
+            # multiples of pi
+            r_ = re(arg)
+            i_ = im(arg)
+            t = (i_/r_).cancel()
```

Comment:
The case `r_ == 0` should be handled somewhere.

Reply:
Thanks for reviewing!

Indeed, explicitly handling `r_==0` would be cleaner (I'll modify the code.). Currently `i_/r_ = zoo` for `arg=5*I` and `zoo` isn't in the table, so it falls through to the next code block, which handles it.

It may be best to also change the order of the code blocks, i.e. put the exact values code at the end of `eval()`. That way most of the purely imaginary arguments will already have been handled (at least rational or infinite multiples of `I`).  And that leaves us with the question of how much *automatic* simplification/evaluation is actually wanted. I'm not sure the following would always be appreciated: `log(I*sqrt(2)) = log(sqrt(2)) + I*pi/2`. (Result arguably more complicated than the input.) But then again it's already implemented for rational coefficients.

With the current PR code, [one of the test cases](https://github.com/sympy/sympy/blob/a2d65f2a9c6c774404d602a941c09008e100025a/sympy/series/tests/test_formal.py#L276) is failing rather spectacularly (overly complex result) and the only relevant `log` calls seem to be `log(1+I)` and `log(1-I)`.
I didn't notice before pushing because it's one of the slow tests that are skipped by `test()`.
I'll try to track it down.

Reply:
I think both issues should now be resolved with the latest commits.